src/main/java/games/strategy/triplea/attachments/UnitAttachment.java

```diff
@@ -434,6 +436,46 @@ public void resetCanBeCapturedOnEnteringBy() {
     m_canBeCapturedOnEnteringBy = new ArrayList<>();
   }
 
+  /**
+   * Adds to, not sets. Anything that adds to instead of setting needs a clear function as well.
+   */
+  @GameProperty(xmlProperty = true, gameProperty = true, adds = true)
+  public void setWhenHitPointsDamagedChangesInto(final String value) throws GameParseException {
+    final String[] s = value.split(":");
+    if (s.length != 3) {
+      throw new GameParseException(
+          "setWhenHitPointsDamagedChangesInto must have hitpoints:keepAttributes:unitType " + thisErrorMsg());
+    }
+    final UnitType unitType = getData().getUnitTypeList().getUnitType(s[2]);
+    if (unitType == null) {
+      throw new GameParseException("setWhenHitPointsDamagedChangesInto: No unit type: " + s[2] + thisErrorMsg());
+    }
+    m_whenHitPointsDamagedChangesInto.put(getInt(s[0]), Tuple.of(getBool(s[1]), unitType));
+  }
+
+  @GameProperty(xmlProperty = true, gameProperty = true, adds = false)
+  public void setWhenHitPointsDamagedChangesInto(final Map<Integer, Tuple<Boolean, UnitType>> value) {
+    m_whenHitPointsDamagedChangesInto = value;
+  }
+
+  /**
+   * Can remove null check and this comment for next incompatible release.
+   */
+  public Map<Integer, Tuple<Boolean, UnitType>> getWhenHitPointsDamagedChangesInto() {
+    if (m_whenHitPointsDamagedChangesInto == null) {
```

Comment:
Do you need a similar `null` check in `clearWhenHitPointsDamagedChangesInto()` and `setWhenHitPointsDamagedChangesInto(String)`?  Or will those code paths not be hit after deserializing an old save game?

Reply:
Fairly sure those shouldn't be hit.